src/bundle/Resources/public/js/scripts/core/suggestion.taggify.js

```diff
@@ -0,0 +1,124 @@
+                        },
+                        limit: 10,
+                        offset: 0,
+                    },
+                },
+            });
+        }
+
+        createSuggestionsRequest(body) {
+            return new Request('/api/ibexa/v2/views', {
+                method: 'POST',
+                headers: {
+                    Accept: 'application/vnd.ibexa.api.View+json; version=1.1',
+                    'Content-Type': 'application/vnd.ibexa.api.ViewInput+json; version=1.1',
+                    'X-Siteaccess': this.siteaccess,
+                    'X-CSRF-Token': this.token,
+                },
+                body,
+                mode: 'same-origin',
+                credentials: 'same-origin',
+            });
+        }
+
+        getSuggestions(query) {
+            const body = this.createSuggestionsRequestBody(query);
+            const request = this.createSuggestionsRequest(body);
+
+            fetch(request)
+                .then(ibexa.helpers.request.getJsonFromResponse)
+                .then(this.getItemsFromResponse)
+                .then(this.renderSuggestionsList)
+                .catch(ibexa.helpers.notification.showErrorNotification);
+        }
+
+        getItemsFromResponse(response) {
+            return response.View.Result.searchHits.searchHit.map((hit) => hit.value.Content);
+        }
+
+        renderSuggestionsList(items) {
+            const fragment = doc.createDocumentFragment();
+
+            items.forEach((item) => {
+                const listItemNode = this.renderSuggestionItem(item);
+
+                listItemNode.addEventListener(
+                    'click',
+                    ({ currentTarget }) => {
+                        this.addTag(currentTarget.innerHTML, item);
+                        this.hideSuggestionsList();
+
+                        this.inputNode.value = '';
+                    },
+                    false,
+                );
+
+                fragment.append(listItemNode);
+            });
+
+            this.suggestionsListNode.innerHTML = '';
+            this.suggestionsListNode.append(fragment);
+
+            this.showSuggestionsList();
+        }
+
+        renderSuggestionItem(item) {
+            const itemTemplate = this.suggestionsListNode.dataset.template;
+            const renderedTemplate = itemTemplate.replace('{{ name }}', item.TranslatedName);
+            const container = doc.createElement('div');
+
+            container.innerHTML = '';
+            container.insertAdjacentHTML('beforeend', renderedTemplate);
+
+            return container.querySelector('div');
+        }
+
+        handleInputKeyUp(event) {
+            super.handleInputKeyUp(event);
+
+            if (this.isAcceptKeyPressed(event.key)) {
+                this.hideSuggestionsList();
+
+                return;
+            }
+
+            if (this.inputNode.value.length > 3) {
```

Comment:
Put 3 to some const at the begininng to remove this magic number, I guess that's min number of characters needed to perform search?